### ISSUE TITLE
libpthread.so not found when linking OpenMP to CUDA lib using conda-build

### ISSUE THREAD

**Issue by gschramm** [conda/conda-build#4257]
### Actual Behavior

When running conda-build on my [project](https://github.com/gschramm/parallelproj/tree/master/recipe) the build fails at the very end when linking the shared CUDA library (complete error message see below).

The system fails to find libpthread.so.0 and libpthread_nonshared.a. I think this is because the path of these libs are not correctly passed by cmake to the linker [here](https://github.com/gschramm/parallelproj/blob/005400516339f84678af3257d021dcfaa0ffb93c/cuda/CMakeLists.txt#L5).
To me, it seems that it is looking for the system libs instead of using the libs that are present in the sysroot of my conda environment.
Moreover, I don't understand why my system g++ is used to link instead of conda's g++

--- 
```
[100%] Linking CUDA shared library libparallelproj_cuda.so
cd $SRC_DIR/cuda && $BUILD_PREFIX/bin/cmake -E cmake_link_script CMakeFiles/parallelproj_cuda.dir/link.txt --verbose=2
/usr/bin/g++ -fPIC -Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--disable-new-dtags -Wl,--gc-sections -Wl,-rpath,$PREFIX/lib -Wl,-rpath-link,$PREFIX/lib -L$PREFIX/lib -shared -Wl,-soname,libparallelproj_cuda.so.0.8.0 -o libparallelproj_cuda.so.0.8.0 CMakeFiles/parallelproj_cuda.dir/src/joseph3d_back_cuda.cu.o CMakeFiles/parallelproj_cuda.dir/src/joseph3d_back_tof_lm_cuda.cu.o CMakeFiles/parallelproj_cuda.dir/src/joseph3d_back_tof_sino_cuda.cu.o CMakeFiles/parallelproj_cuda.dir/src/joseph3d_fwd_cuda.cu.o CMakeFiles/parallelproj_cuda.dir/src/joseph3d_fwd_tof_lm_cuda.cu.o CMakeFiles/parallelproj_cuda.dir/src/joseph3d_fwd_tof_sino_cuda.cu.o CMakeFiles/parallelproj_cuda.dir/src/ray_cube_intersection_cuda.cu.o CMakeFiles/parallelproj_cuda.dir/src/tof_utils_cuda.cu.o CMakeFiles/parallelproj_cuda.dir/src/utils_cuda.cu.o CMakeFiles/parallelproj_cuda.dir/cmake_device_link.o   -L$BUILD_PREFIX/x86_64-conda-linux-gnu/sysroot/usr/lib  -Wl,-rpath,$BUILD_PREFIX/x86_64-conda-linux-gnu/sysroot/usr/lib: $PREFIX/lib/libgomp.so -lpthread -lcudadevrt -lcudart_static -lrt -lpthread -ldl  -L"/users/nexuz/gschra2/.conda/envs/ppp-gpu/lib64/stubs" -L"/users/nexuz/gschra2/.conda/envs/ppp-gpu/lib64"
/usr/bin/ld: cannot find /lib64/libpthread.so.0
/usr/bin/ld: cannot find /usr/lib64/libpthread_nonshared.a
collect2: error: ld returned 1 exit status
make[2]: *** [cuda/CMakeFiles/parallelproj_cuda.dir/build.make:263: cuda/libparallelproj_cuda.so.0.8.0] Error 1
make[1]: *** [CMakeFiles/Makefile2:169: cuda/CMakeFiles/parallelproj_cuda.dir/all] Error 2
make[2]: Leaving directory '$SRC_DIR'
make[1]: Leaving directory '$SRC_DIR'
make: *** [Makefile:156: all] Error 2
```
### Expected Behavior

1. link to libthread in sysroot 
2. use conda's compiler to link

### Steps to Reproduce

```
git clone https://github.com/gschramm/parallelproj
conda-build .
```

##### `Output of conda info`

```
    active environment : ppp-gpu
    active env location : /users/nexuz/gschra2/.conda/envs/ppp-gpu
            shell level : 2
       user config file : /users/nexuz/gschra2/.condarc
 populated config files : /users/nexuz/gschra2/.condarc
          conda version : 4.10.1
    conda-build version : not installed
         python version : 3.8.5.final.0
       virtual packages : __cuda=11.2=0
                          __linux=5.4.0=0
                          __glibc=2.31=0
                          __unix=0=0
                          __archspec=1=x86_64
       base environment : /opt/miniconda3  (read only)
      conda av data dir : /opt/miniconda3/etc/conda
  conda av metadata url : https://repo.anaconda.com/pkgs/main
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /opt/miniconda3/pkgs
                          /users/nexuz/gschra2/.conda/pkgs
       envs directories : /users/nexuz/gschra2/.conda/envs
                          /opt/miniconda3/envs
               platform : linux-64
             user-agent : conda/4.10.1 requests/2.25.1 CPython/3.8.5 Linux/5.4.0-71-generic ubuntu/20.04.2 glibc/2.31
                UID:GID : 56322:1271
             netrc file : None
           offline mode : False

```

**Comment by KrisThielemans** [comment]
To me it looks like a problem due to `-Wl,-rpath-link,$PREFIX/lib -L$PREFIX/lib`, and in particular that `$PREFIX` is probably empty when this gets executed.

I guess this comes from
https://github.com/gschramm/parallelproj/blob/0325b370ab7069fe014e1934bb752af460456e87/recipe/cross-linux.cmake#L11-L12
I have no idea why you need the `cross-linux.cmake` by the way.

**Comment by gschramm** [comment]
in included the cross-linux.cmake because of this [article](https://docs.conda.io/projects/conda-build/en/latest/resources/compiler-tools.html#an-aside-on-cmake-and-sysroots) ,
but I must admit that I don't really understand why this is needed.
(cross compiling is right now not the highest priority)

**Comment by KrisThielemans** [comment]
possibly that article is outdated as the [related CMake issue](https://gitlab.kitware.com/cmake/cmake/-/issues/17483) has been closed since CMake 3.13 I believe. If that's the case, that article should probably be updated.

**Comment by KrisThielemans** [comment]
I think this was a glibc bug, now no longer present. See https://github.com/conda-forge/stir-feedstock/issues/101.
